src/pybind/mgr/cephadm/serve.py

```diff
@@ -484,20 +485,39 @@ def matches_network(host):
             # host
             return len(self.mgr.cache.networks[host].get(public_network, [])) > 0
 
+        def virtual_ip_allowed(host):
+            # type: (str) -> str
+            # Verify that it is possible to use Virtual IPs in the host
+            msg = ''
+            try:
+                if self.mgr.cache.facts[host]['kernel_parameters']['net.ipv4.ip_nonlocal_bind'] == '0':
+                    msg = '<net.ipv4.ip_nonlocal_bind> must be set to <1>'
+            except KeyError:
+                msg = 'cephadm not providing the value of <net.ipv4.ip_nonlocal_bind>'
+
+            return msg
+
         ha = HostAssignment(
             spec=spec,
             hosts=self.mgr._hosts_with_daemon_inventory(),
             get_daemons_func=self.mgr.cache.get_daemons_by_service,
-            filter_new_host=matches_network if daemon_type == 'mon' else None,
+            filter_new_host=matches_network if service_type == 'mon' else None,
+            host_validator=virtual_ip_allowed if service_type == 'ha-rgw' else None,
```

Comment:
yeah, that doesn't make sense. can we either use `filter-new-host` OR `host-validator`?